pkg/disruptors/pod_test.go

```diff
@@ -334,7 +394,25 @@ func Test_PodGrpcPFaultInjection(t *testing.T) {
 				executor:  executor,
 			}
 
-			d := newPodDisruptorForTesting(context.TODO(), tc.selector, controller)
+			objs := []runtime.Object{}
+
+			for _, target := range tc.targets {
+				obj := builders.NewPodBuilder(target).
+					WithLabels(tc.selector.Select.Labels).
+					WithNamespace(tc.selector.Namespace).
+					WithContainerPort(DefaultTargetPort).
```

Comment:
I think this is not necessary, is it? I find it confusing we are setting this in this test.

Reply:
You mean container port? It's needed in order to prevent all existing test cases from failing. I can replace `PortValidator` with a dummy one with empty targets array. In this case I can avoid k8s objects mocking entirely but I won't be able to test the case when validator throws an error. Another option would be introducing an interface for PortValidator and have `fakePortValidator` for disruptor tests (as it is for AgentController). This approach will make disruptor tests cleaner and easier to understand  but it will introduce new interface in code.

BTW, I noticed that I forgot to remove `containerPort` in test case struct and I left `expectedCmd` in new tests, I will clean that 👌

Reply:
Ok, I see. I would prefer that the `PodBuilder` provided that default container port instead of requiring to add it on each test, so that `NewPodBuilder(..).Build()` should provide a default pod that can be used in most test cases.

Reply:
Oh, that's indeed better, I overcomplicated it 😅

Reply:
And what do you think about getting rid of mocking k8s objects and mocking PortValidator instead?

Reply:
Fixed. But now I noticed that there is an implicit dependency on default port value between builders (both pod and service) and disruptors packages. It doesn't look harmful now but maybe it's worth refactoring in future

Reply:
Hi @nickandreev do you think the PR is ready from your side? I plan to make a final review and merge it.